deterministic-build-wrappers/rustc

```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+# rustc calculates a unique metadata tag to mangle symbols which includes the
+# actual path to the crate. This breaks our deterministic builds as we depend
```

Comment:
So, is the problem this script is solving basically: something something C++ only supports a global namespace, therefore all symbols need the complete path, so we need to make sure the path is the same no matter when or where this is being built?

Reply:
rustc modifies every symbol (ie function name) with the metadata string so that functions are globally-unique even when you have two different versions of a crate with the same function. That's great, except it also includes the hash of the path to the crate in its metadata hash, so we need to stub it out.